azure-mgmt-monitor/src/main/java/com/microsoft/azure/management/monitor/implementation/MetricDefinitionImpl.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ *
+ * Code generated by Microsoft (R) AutoRest Code Generator.
+ */
+
+package com.microsoft.azure.management.monitor.implementation;
+import com.microsoft.azure.management.apigeneration.LangDefinition;
+import com.microsoft.azure.management.apigeneration.LangMethodDefinition;
+import com.microsoft.azure.management.monitor.AggregationType;
+import com.microsoft.azure.management.monitor.LocalizableString;
+import com.microsoft.azure.management.monitor.MetricAvailability;
+import com.microsoft.azure.management.monitor.MetricCollection;
+import com.microsoft.azure.management.monitor.MetricDefinition;
+import com.microsoft.azure.management.monitor.ResultType;
+import com.microsoft.azure.management.monitor.Unit;
+import org.joda.time.DateTime;
+import org.joda.time.Period;
+import rx.Observable;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.ISODateTimeFormat;
+import rx.functions.Func1;
+
+import java.util.List;
+
+/**
+ * The Azure metric definition entries are of type MetricDefinition.
+ */
+@LangDefinition(ContainerName = "/Microsoft.Azure.Management.Monitor.Fluent.Models")
+class MetricDefinitionImpl
+            implements
+                MetricDefinition,
+                MetricDefinition.Definition {
+
```

Comment:
extend `WrapperImpl<MetricDefinitionInner>` if you decide to extend `MetricDefinition` from `HasInner< MetricDefinitionInner>` based on my previous comment

Reply:
fixed